### PR TITLE
[training] fix: Route batches to standalone MTP stages

### PR DESCRIPTION
**Pull request by yaoyu-33**
# What does this PR do?

Fixes an internally reported standalone MTP pipeline-layout issue in GPT batch routing.

When a custom pipeline layout places an MTP block on its own stage, that stage needs token and position metadata to build the MTP forward inputs. The previous middle-pipeline fast path returned all `None` values, so the standalone MTP stage could not receive the needed inputs.

This change updates GPT batch routing so:

- a pipeline stage that owns an MTP block receives `tokens` and `position_ids`;
- ordinary middle pipeline stages keep the existing all-`None` fast path;
- the standalone loss-only final stage does not receive unnecessary MTP inputs.

# Changelog

- Route GPT batches based on MTP-stage needs in addition to first/last pipeline-stage ownership.
- Preserve the existing non-MTP middle-stage fast path.
- Add targeted unit coverage for standalone MTP stage routing and loss-only final-stage behavior.

# Verification

- `uv run --no-sync pre-commit run --all-files`
- `uv run --no-sync python -m pytest tests/unit_tests/training/test_gpt_step.py -v`
- Targeted multi-GPU standalone-MTP mock pretraining completed 5/5 iterations with MTP loss logged and zero skipped/nan iterations.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci) in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation? No public docs needed.
- [x] Does the PR affect components that are optional to install? No.
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

### CODE REVIEW

File: src/megatron/bridge/training/gpt_step.py [hunk 1]
```diff
@@ -57,6 +59,66 @@ def _middle_pp_stage_needs_batch(cfg: ConfigContainer) -> bool:
     return uses_custom_attention_mask or _uses_packed_sequence_metadata(cfg)
 
 
+def _get_pp_group_rank(pp_group) -> int:
```
Comment (yaoyu-33) [inline src/megatron/bridge/training/gpt_step.py]: use get_pg_rank and size util from mcore.

Comment (yaoyu-33) [reply src/megatron/bridge/training/gpt_step.py]: Fixed in 3e8bd9a44ef2b62d648829767b1abfd8c5a9545b: switched PP layout rank/size lookup to MCore get_pg_rank/get_pg_size and removed the local helper wrappers.

File: src/megatron/bridge/training/gpt_step.py [hunk 2]
```diff
@@ -191,12 +253,15 @@ def get_batch(
...
+    include_mtp_inputs = use_mtp and _current_pp_stage_needs_mtp_inputs(
+        cfg, pg_collection=pg_collection, is_last=is_last
+    )
+    if is_middle and not include_full_batch_fields and not include_mtp_inputs:
         return None, None, None, None, None, None, None, None, None, None
 
     batch = get_batch_from_iterator(
         data_iterator,
-        use_mtp,
+        include_mtp_inputs,
```
[the remark this answers is not in the document]
Comment (yaoyu-33) [reply src/megatron/bridge/training/gpt_step.py]: Fixed in 3e8bd9a44ef2b62d648829767b1abfd8c5a9545b: renamed the parameter to include_mtp_inputs and updated the docstring/call site to match the per-stage routing semantics.

**Review by cuichenx — APPROVED** [review]
LGTM
